src/subscription_manager/api.py

```diff
@@ -0,0 +1,83 @@
+#
+# Copyright (c) 2015 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public
+# License as published by the Free Software Foundation; either version
+# 2 of the License (GPLv2) or (at your option) any later version.
+# There is NO WARRANTY for this software, express or implied,
+# including the implied warranties of MERCHANTABILITY,
+# NON-INFRINGEMENT, or FITNESS FOR A PARTICULAR PURPOSE. You should
+# have received a copy of GPLv2 along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+#
+
+"""
+Class and methods in this module are provided for use by external applications with
+the exception of methods beginning with an '_'.
+
+All reasonable efforts will be made to maintain compatibility.
+"""
+import fnmatch
+
+from subscription_manager import logutil
+logutil.init_logger()
+
+from subscription_manager.injectioninit import init_dep_injection
+init_dep_injection()
+
+import subscription_manager.injection as inj
+from subscription_manager.repolib import RepoActionInvoker, RepoFile
+
+
+def _set_enable_for_repositories(setting, *repo_ids):
+    invoker = RepoActionInvoker()
+    repos = invoker.get_repos()
+    repos_to_change = []
+
+    for r in repo_ids:
+        matches = set([repo for repo in repos if fnmatch.fnmatch(repo.id, r)])
```

Comment:
> https://docs.python.org/2/library/fnmatch.html#fnmatch.filter instead?

`fnmatch.filter` doesn't quite fit because instead of 

```python
repo for repo in repos if fnmatch.fnmatch(repo, some_id)
```

I want 

```python
repo for repo in repos if fnmatch.fnmatch(repo.id, some_id)
```

So I don't know how to get `filter` to filter based on an attribute of the list objects and then return the list objects themselves without doing some quasi-Schwartzian transform type operation that would end up being harder to read.